platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -97,21 +97,22 @@ private static void addLocations(ProblemSpec spec, Diagnostic<? extends JavaFile
         String resourceName = diagnostic.getSource() != null ? getPath(diagnostic.getSource()) : null;
         int line = clampLocation(diagnostic.getLineNumber());
         int column = clampLocation(diagnostic.getColumnNumber());
+        int position = clampLocation(diagnostic.getPosition());
         int start = clampLocation(diagnostic.getStartPosition());
         int end = clampLocation(diagnostic.getEndPosition());
 
         // We only set the location if we have a resource to point to
         if (resourceName != null) {
             spec.fileLocation(resourceName);
             // If we know the line ...
-            if (line > 0) {
+            if (0 < line) {
                 // ... and the column ...
-                if (column > 0) {
+                if (0 < column) {
                     // ... and we know how long the error is (i.e. end - start)
-                    // (end should be greater than start, so we can prevent negative lengths)
-                    if (0 < start && start <= end) {
+                    // (documentation says that getEndPosition() will be NOPOS iff getPosition() is NOPOS)
```

Comment:
iff -> if

Reply:
Maybe I went to mathematical here, official doc says "if and only if".